examples/seismic/tutorials/04_dask.ipynb

```diff
@@ -178,41 +180,40 @@
     "        'observed' data using the real model. For a real case\n",
     "        the worker would be reading seismic data from disk.\n",
     "    \"\"\"\n",
-    "    true_model = get_true_model()\n",
-    "    shot_id = param['shot_id']\n",
-    "    \n",
-    "    src_coordinates = np.empty((1, len(param['shape'])))\n",
-    "    src_coordinates[0, :] = [30, param['shot_id']*1000./(param['nshots']-1)]\n",
+    "    # Reconstruct objects\n",
+    "    with open(\"arguments.pkl\", \"rb\") as cp_file:\n",
+    "        cp = pickle.load(cp_file)\n",
+    "        \n",
+    "    op = cp['operator']\n",
+    "    geometry = cp['geometry']\n",
+    "    # source position changes according to the index\n",
+    "    shot_id=param['shot_id']\n",
     "    \n",
-    "    # Number of receiver locations per shot.\n",
-    "    nreceivers = 101\n",
-    "\n",
-    "    # Set up receiver data and geometry.\n",
-    "    rec_coordinates = np.empty((nreceivers, len(param['shape'])))\n",
-    "    rec_coordinates[:, 1] = np.linspace(0, true_model.domain_size[0], num=nreceivers)\n",
-    "    rec_coordinates[:, 0] = 980. # 20m from the right end\n",
+    "    geometry.src_positions[0,:]=[30, shot_id*1000./(param['nshots']-1)]\n",
+    "    true_d, _, _ = op.forward()\n",
+    "    dump_shot_data(shot_id, true_d, geometry)\n",
     "\n",
+    "def generate_shotdata(src_coor,rec_coor):\n",
+    "    true_model = get_true_model()\n",
     "    # Geometry \n",
-    "    geometry = AcquisitionGeometry(true_model, rec_coordinates, src_coordinates,\n",
+    "    geometry = AcquisitionGeometry(true_model, rec_coor, src_coor,\n",
     "                                   param['t0'], param['tn'], src_type='Ricker',\n",
     "                                   f0=param['f0'])\n",
-    "    # Set up solver.\n",
     "    solver = AcousticWaveSolver(true_model, geometry, space_order=4)\n",
```

Comment:
This the only place where the `solver` is created. This same `solver` is used in the `generated_shotdata_i` and `fwi_gradient_i` functions.  It could be passed as an argument to the functions but I did not want to modify much the structure of the notebook so I created a dictionary with the `geometry` and the `solver` and then I used the `pickle` module. An “unpickling” operation is performed in the functions in order to use the `solver`.

Reply:
In that case, firstly can we have the comment back? No reason to remove an explanatory comment, I guess?

Reply:
Yes, the comment has to stay.